zebra-state/src/service/finalized_state/disk_format/chain.rs

```diff
@@ -39,42 +43,54 @@ impl FromDisk for ValueBalance<NonNegative> {
 // https://docs.rs/bincode/1.3.3/bincode/config/index.html#options-struct-vs-bincode-functions
 
 #[derive(serde::Serialize, serde::Deserialize)]
-struct HistoryTreeParts {
-    network: Network,
+pub struct HistoryTreeParts {
+    network: NetworkKind,
     size: u32,
     peaks: BTreeMap<u32, zcash_history::Entry>,
     current_height: Height,
 }
 
-impl IntoDisk for NonEmptyHistoryTree {
+impl HistoryTreeParts {
+    /// Converts [`HistoryTreeParts`] to a [`NonEmptyHistoryTree`].
+    pub(crate) fn with_network(
+        self,
+        network: &Network,
+    ) -> Result<NonEmptyHistoryTree, HistoryTreeError> {
+        assert_eq!(
+            self.network,
+            network.kind(),
+            "history tree network kind should match current network"
+        );
+
+        NonEmptyHistoryTree::from_cache(network, self.size, self.peaks, self.current_height)
+    }
+}
+
+impl From<&NonEmptyHistoryTree> for HistoryTreeParts {
+    fn from(history_tree: &NonEmptyHistoryTree) -> Self {
+        HistoryTreeParts {
+            network: history_tree.network().kind(),
+            size: history_tree.size(),
+            peaks: history_tree.peaks().clone(),
+            current_height: history_tree.current_height(),
+        }
+    }
+}
+
+impl IntoDisk for HistoryTreeParts {
     type Bytes = Vec<u8>;
 
     fn as_bytes(&self) -> Self::Bytes {
-        let data = HistoryTreeParts {
```

Comment:
`Network` is now always converted into a `NetworkKind` before being serialized anyway, but we still need an `IntoDisk` impl for `HistoryTreeParts` since we can't deserialize the full network on history tree and using `HistoryTreeParts` in the column family type requires both `IntoDisk` and `FromDisk` impls.

Check if it's possible to use `NetworkKind` in the `HistoryTree` type to revert some of these changes?